app/database/operator/server_data_operator/handlers/thread_in_team.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Q} from '@nozbe/watermelondb';
+
+import {Database} from '@constants';
+import {transformThreadInTeamRecord} from '@database/operator/server_data_operator/transformers/thread';
+import {getRawRecordPairs} from '@database/operator/utils/general';
+import {HandleThreadInTeamArgs} from '@typings/database/database';
+
+import type ThreadInTeamModel from '@typings/database/models/servers/team_thread';
+
+export interface ThreadInTeamHandlerMix {
+    handleThreadInTeam: ({threadsMap, prepareRecordsOnly}: HandleThreadInTeamArgs) => Promise<ThreadInTeamModel[]>;
+}
+
+const {THREADS_IN_TEAM} = Database.MM_TABLES.SERVER;
+
+const ThreadInTeamHandler = (superclass: any) => class extends superclass {
+    handleThreadInTeam = async ({threadsMap, prepareRecordsOnly = false}: HandleThreadInTeamArgs): Promise<ThreadInTeamModel[]> => {
+        if (!Object.keys(threadsMap).length) {
+            return [];
+        }
+
+        const create: ThreadInTeam[] = [];
+        const teamIds = Object.keys(threadsMap);
+        for await (const teamId of teamIds) {
+            const chunks = (await this.database.get(THREADS_IN_TEAM).query(
+                Q.where('team_id', teamId),
+            ).fetch()) as ThreadInTeamModel[];
```

Comment:
```suggestion
            const chunks = (await this.database.get<ThreadInTeamModel>(THREADS_IN_TEAM).query(
                Q.where('team_id', teamId),
            ).fetch());
```

Reply:
for some dark reason this does not work..